### DISCUSSION TITLE
micropython with Adafruit Grand Central M4 Express

### DISCUSSION THREAD

**Discussion by AlessandroCortese** [micropython/micropython#11021] [SAMD]
Good morning,

Has anyone ever used micropython with Adafruit Grand Central M4 Express?

**Comment by robert-hh** [comment]
@AlessandroCortese  I have responded to your question at the Pycom forum:
There is no ready port for that board. But idf you have one, you can find a set of firmware images at https://github.com/robert-hh/Shared-Stuff/tree/master/samd_firmware. As a start, you could try the GENERIC_SAMD51J20_firmware.uf2 firmware. It makes the least assumptions about the hardware, so it does not use the crystal or the QSPI flash. But that could easily added, if the Generic firmware works.

**Reply by robert-hh** [reply]
I added a trial version for the Grand Central Express M4.

**Comment by graeme-winter** [comment]
For any later reader... Yes, dormant branch (which @robert-hh contributed to) at

https://github.com/micropython/micropython/compare/master...winter-special-projects:micropython:adafruit-grand-central

Probably bringing this back wouldn't be imposssible as I have the board here. It worked quite well but the frustration of getting it actually merged was substantial and I gave up at the time. @robert-hh did (very reasonably) ask me a while back if I intended to resurrect this.
